src/main/java/tech/stackable/hbase/opa/OpaAclChecker.java

```diff
@@ -0,0 +1,190 @@
+    }
+  }
+
+  public OpaAclChecker(
+      boolean authorizationEnabled, String opaPolicyUrl, boolean dryRun, CacheConfig cc) {
+    this.authorizationEnabled = authorizationEnabled;
+    this.dryRun = dryRun;
+    this.aclCache =
+        cc.enabled
+            ? Optional.of(
+                Caffeine.newBuilder()
+                    .expireAfterWrite(cc.ttlSeconds, TimeUnit.SECONDS)
+                    .maximumSize(cc.maxSize)
+                    .build())
+            : Optional.empty();
+
+    this.json =
+        new ObjectMapper()
+            .configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false)
+            .configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false)
+            .setVisibility(PropertyAccessor.ALL, JsonAutoDetect.Visibility.NONE)
+            .setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.PUBLIC_ONLY)
+            .setVisibility(PropertyAccessor.GETTER, JsonAutoDetect.Visibility.PUBLIC_ONLY);
+
+    if (authorizationEnabled) {
+      if (opaPolicyUrl == null) {
+        throw new OpaException.UriMissing(OpenPolicyAgentAccessController.OPA_POLICY_URL_PROP);
+      }
+
+      try {
+        this.opaUri = URI.create(opaPolicyUrl);
+      } catch (Exception e) {
+        throw new OpaException.UriInvalid(opaUri, e);
+      }
+    }
+  }
+
+  public void checkPermissionInfo(User user, TableName table, Permission.Action action)
+      throws AccessControlException {
+    OpaAllowQuery query =
+        new OpaAllowQuery(new OpaAllowQuery.OpaAllowQueryInput(user.getUGI(), table, action));
+    this.checkPermissionInfo(query);
+  }
+
+  public void checkPermissionInfo(User user, String namespace, Permission.Action action)
+      throws AccessControlException {
+    OpaAllowQuery query =
+        new OpaAllowQuery(new OpaAllowQuery.OpaAllowQueryInput(user.getUGI(), namespace, action));
+    this.checkPermissionInfo(query);
+  }
+
+  public void checkPermissionInfo(OpaAllowQuery query) throws AccessControlException {
+    if (!this.authorizationEnabled) {
+      return;
+    }
+
+    String body;
+    try {
+      body = json.writeValueAsString(query);
+    } catch (JsonProcessingException e) {
+      throw new OpaException.SerializeFailed(e);
+    }
+
+    String prettyPrinted;
+    try {
+      prettyPrinted = json.writerWithDefaultPrettyPrinter().writeValueAsString(query);
+    } catch (JsonProcessingException e) {
+      LOG.error(
+          "Could not pretty print the following request body (printing raw version instead): {}",
+          body);
+      throw new OpaException.SerializeFailed(e);
+    }
+
+    LOG.info("Request body:\n{}", prettyPrinted);
+    if (this.dryRun) {
+      LOG.info("Dry run request: omitting call.");
+      return;
+    }
+
+    if (aclCache.isPresent()) {
+      final Boolean result = aclCache.get().getIfPresent(body);
+      if (result != null) {
+        if (result) {
+          LOG.info("Permission exists in OPA-policy-cache, by-passing policy call");
```

Comment:
not info

(I'll stop commenting on the log.info related things now, I'm sure there are more of this here)

Reply:
Will be addressed in https://github.com/stackabletech/hbase-opa-authorizer/issues/2